File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs8/EncryptedPrivateKeyInfo.java
package aladdin.asn1.iso.pkcs.pkcs8;
import aladdin.asn1.*; 
import aladdin.asn1.iso.*; 
import java.io.*;

//	EncryptedPrivateKeyInfo ::= SEQUENCE {
//		encryptionAlgorithm	AlgorithmIdentifier,
//		encryptedData		OCTET STRING
//	}

public final class EncryptedPrivateKeyInfo extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(OctetString		.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public EncryptedPrivateKeyInfo(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public EncryptedPrivateKeyInfo(AlgorithmIdentifier encryptionAlgorithm, 
		OctetString encryptedData) 
	{
		super(info, encryptionAlgorithm, encryptedData); 
	}
	public final AlgorithmIdentifier	encryptionAlgorithm() { return (AlgorithmIdentifier	)get(0); } 
	public final OctetString            encryptedData	   () { return (OctetString         )get(1); }
}

File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS08/EncryptedPrivateKeyInfo.cs
using System;
using System.Runtime.Serialization;

//	EncryptedPrivateKeyInfo ::= SEQUENCE {
//		encryptionAlgorithm	AlgorithmIdentifier,
//		encryptedData		OCTET STRING
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS8
{
	[Serializable]
	public class EncryptedPrivateKeyInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<OctetString		>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected EncryptedPrivateKeyInfo(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public EncryptedPrivateKeyInfo(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public EncryptedPrivateKeyInfo(AlgorithmIdentifier encryptionAlgorithm, 
			OctetString encryptedData) : base(info, encryptionAlgorithm, encryptedData) {}

		public AlgorithmIdentifier	EncryptionAlgorithm	{ get { return (AlgorithmIdentifier	)this[0]; } } 
		public OctetString			EncryptedData		{ get { return (OctetString			)this[1]; } }
	}
}
